alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ExpressionManager.java

```diff
@@ -81,31 +94,91 @@
     @Inject
     private AlertsService alerts;
 
+    /**
+     * Access to the manager of the caches used for the partition services.  This is used to inspect cluster
+     * topology and ensure we only run the external alerter on one node (the coordinator). Otherwise
+     * each node would execute the same work, resulting in duplicate alerts.
+     */
+    @Resource(lookup = "java:jboss/infinispan/container/hawkular-alerts")
+    private EmbeddedCacheManager cacheManager;
+
     @PostConstruct
     public void init() {
-        log.debug("Initializing Hawkular Alerts-Metrics Manager...");
+        // Cache manager has an active transport (i.e. jgroups) when is configured on distributed mode
+        distributed = cacheManager.getTransport() != null;
+
+        if (distributed) {
+            topologyChangeListener = new TopologyChangeListener();
+            cacheManager.addListener(topologyChangeListener);
+        }
+
+        processTopologyChange();
+    }
+
+    // When a node is joining/leaving the cluster the coordinator node could change
+    @Listener
+    public class TopologyChangeListener {
+        @ViewChanged
+        public void onTopologyChange(ViewChangedEvent cacheEvent) {
+            processTopologyChange();
+        }
+    }
+
+    private void processTopologyChange() {
+        boolean currentCoordinator = coordinator;
+        coordinator = distributed ? cacheManager.isCoordinator() : true;
+
+        if (coordinator && !currentCoordinator) {
+            start();
```

Comment:
That's a good observation and if it is an issue I think it will probably be solved later, when we convert to using Metrics' job scheduler (see the Jira for a comment about the two considered solutions).  For now, hopefully JGroups/ISPN informs listeners in a timely manner, because the stop() operation is quite fast, just issues any necessary cancels.   Although possible, if there is brief overlap, it is unlikely to generate duplicate alerts given that alerts should be rare.